playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java

```diff
@@ -0,0 +1,34 @@
+package com.microsoft.playwright.junit.impl;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.microsoft.playwright.Playwright;
+import com.microsoft.playwright.impl.PlaywrightImpl;
+import com.microsoft.playwright.options.ViewportSize;
+
+class DeviceDescriptor {
+  public String userAgent;
+  public ViewportSize viewport;
+  public Double deviceScaleFactor;
+  public Boolean isMobile;
+  public Boolean hasTouch;
+  public String defaultBrowserType;
+
+  static DeviceDescriptor findByName(Playwright playwright, String name) {
+    JsonArray devices = ((PlaywrightImpl) playwright).deviceDescriptors();
```

Comment:
@yury-s  this is returning null for me and so a null pointer exception is being thrown.

Reply:
![image](https://github.com/microsoft/playwright-java/assets/1053584/5a9f5403-9698-4656-ab4d-8d793bfbd3e3)

Reply:
Hmm, that's odd. Is `options` in `options.deviceName` null or where does it come from?

Reply:
I believe `PlaywrightImpl.deviceDescriptors()` is returning null so the for loop throws a null pointer exception:

![image](https://github.com/microsoft/playwright-java/assets/1053584/8bf61adf-7be2-40fe-a5ed-4de34ffd9879)

Reply:
Hmm, do you have a test where this is happening? This should never happen if playwright is initialized.

Reply:
Ok, I assume it is just the new test. It passes on the bots and locally. Could it be that you have some stale driver? We changed the way device descriptors pushed to the client some time ago. Try downloading [current driver](https://github.com/microsoft/playwright-java/blob/main/CONTRIBUTING.md#updating-driver-version) and see if it still fails.

Reply:
Just running `./scripts/download_driver_for_all_platforms.sh -f` and rebuilding should be sufficient

Reply:
it was failing on main for me too but you figured it out.  downloaded the current driver and it's all good.  apologies for that.  i'll make running that command part of my workflow.